### ISSUE TITLE
[Practice Exam Question] Doubts about the class diagram on sub-activities

### ISSUE THREAD

**Issue by damithc** [nus-cs2103-AY2324S2/forum#915]
[SPOILER ALERT] This post reveals the answer to one of the practice exam questions. If you haven't done the practice exam yet, you might want to stop reading at this poing.

Given below are some questions received from another student. Posting here for discussion:

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/1673303/a5c5afb5-d788-46ff-aa64-650d75e36f16)

1. Why the answer is (e)? My original answer was (b) as there is no mention of the sub-activities...
1. Regarding this question about activity keeping track of sub-activities, I don't get why there is a composition relation... 
1. I believe that deletion of the main activity keeping track of the sub-activity does not mean that the sub-activity would get deleted?
1. What do arrows mean in composition ? I notice that some composition have arrows while some don't, is there any difference?

**Comment by itstrueitstrueitsrealitsreal** [comment]
Is the answer E because the association between Activity and itself is shown twice, once as an attribute of activity and another time using the arrow?

**Comment by baskargopinath** [comment]
@itstrueitstrueitsrealitsreal i agree with u but why is C correct if parts is not mentioned anywhere

**Comment by baskargopinath** [comment]
> 1. Why the answer is (e)? My original answer was (b) as there is no mention of the sub-activities...

Isn't sub-activities mentioned in the heading

**Comment by tankh99** [comment]
Is the reason why E is correct because the sub-activity relationship is being expressed as a line and an attribute at the same time? Or is it because it is showing the sub-activity attribute with an invalid name?

**Comment by tankh99** [comment]
@damithc  bump

**Comment by damithc** [comment]
> Is the answer E because the association between Activity and itself is shown twice, once as an attribute of activity and another time using the arrow?

This is the correct reason.

> * Regarding this question about activity keeping track of sub-activities, I don't get why there is a composition relation...
> * I believe that deletion of the main activity keeping track of the sub-activity does not mean that the sub-activity would get deleted?

_Delete cascading_ is an effect of composition, not the cause. More info [here](https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/modeling.html#modeling-composition:~:text=Cascading%20deletion%20alone%20is%20not%20sufficient%20for%20composition)

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/1673303/3e116de1-2722-4da5-a979-9a5f37898ddd)

Also relevant, Q3 of week 9 briefing:
![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/1673303/9d0748f5-dfc6-4fa3-825d-74b9dbb2a27d)


> 4\. What do arrows mean in composition ? I notice that some composition have arrows while some don't, is there any difference?

Arrow indicates navigability. It's independent of composition.
